Mathlib/FieldTheory/Minpoly/IsConjRoot.lean

```diff
@@ -0,0 +1,310 @@
+/-
+Copyright (c) 2024 Jiedong Jiang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jiedong Jiang
+-/
+import Mathlib.FieldTheory.Minpoly.Basic
+import Mathlib.FieldTheory.Adjoin
+import Mathlib.FieldTheory.Normal
+
+/-!
+# Conjugate roots
+
+Given two elements `x` and `y` of some `K`-algebra, these two elements are *conjugate roots*
+over `K` if they have the same minimal polynomial over `K`.
+
+## Main definitions
+
+* `IsConjRoot`: `IsConjRoot K x y` means `y` is a conjugate root of `x` over `K`.
+
+## Main results
+
+* `isConjRoot_iff_exists_algEquiv`: Let `L / K` be a normal field extension. For any two elements
+`x` and `y` in `L`, `IsConjRoot K x y` is equivalent to the existence of an algebra equivalence
+`σ : L ≃ₐ[K] L` such that `y = σ x`.
+* `not_mem_iff_exists_ne_and_isConjRoot`: Let `L / K` be a field extension. If `x` is a separable
+element over `K` and the minimal polynomial of `x` splits in `L`, then `x` is not in the `K` iff
+there exists a different conjugate root of `x` in `L` over `K`.
+
+## TODO
+Prove `IsConjRoot.smul`, if `x` and `y` are conjugate roots, then so are `r • x` and `r • y`.
+
+## Tags
+conjugate root, minimal polynomial
+-/
+
+
+open Polynomial minpoly IntermediateField
+
+variable {R K L S A B: Type*} [CommRing R] [CommRing S] [Ring A] [Ring B] [Field K] [Field L]
+variable [Algebra R S] [Algebra R A] [Algebra R B]
+variable [Algebra K S] [Algebra K L] [Algebra K A] [Algebra L S]
+
+variable (R) in
+/--
+We say that `y` is a conjugate root of `x` over `K` if the minimal polynomial of `x` is the
+same as the minimal polynomial of `y`.
+-/
+def IsConjRoot (x y : A) : Prop := (minpoly R x) = (minpoly R y)
+
+/--
+The definition of conjugate roots.
+-/
+theorem isConjRoot_def {x y : A} : IsConjRoot R x y ↔ minpoly R x = minpoly R y := Iff.rfl
+
+namespace IsConjRoot
+
+/--
+Every element is a conjugate root of itself.
+-/
+theorem refl {x : A} : IsConjRoot R x x := rfl
+
+/--
+If `y` is a conjugate root of `x`, then `x` is also a conjugate root of `y`.
+-/
+theorem symm {x y : A} (h : IsConjRoot R x y) : IsConjRoot R y x := Eq.symm h
+
+/--
+If `y` is a conjugate root of `x` and `z` is a conjugate root of `y`, then `z` is a conjugate
+root of `x`.
+-/
+theorem trans {x y y': A} (h₁ : IsConjRoot R x y) (h₂ : IsConjRoot R y y') :
+    IsConjRoot R x y' := Eq.trans h₁ h₂
```

Comment:
[#6718](https://github.com/leanprover-community/mathlib4/pull/6718/files#r1816397133) needs the Setoid so you may as well add it here.

Reply:
Added. Thank you!